samza-core/src/main/java/org/apache/samza/config/JobConfig.java

```diff
@@ -479,6 +484,10 @@ public boolean getContainerHeartbeatMonitorEnabled() {
     return getBoolean(CONTAINER_HEARTBEAT_MONITOR_ENABLED, CONTAINER_HEARTBEAT_MONITOR_ENABLED_DEFAULT);
   }
 
+  public boolean getElasticityCheckpointEnabled() {
+    return getBoolean(JOB_ELASTICITY_CHECKPOINTS_ENABLED, DEFAULT_JOB_ELASTICITY_CHECKPOINTS_ENABLED);
```

Comment:
Can we just use getElasticityEnabled() inplace of this?
That is, whenever elasticity factor = 1, we default to false?

Reply:
so the reason for introducing this new config is the following scenario

1. First, job has no elasticity aka factor = 1 by default and checkpoints are written with taskName like "Partition 0"
2. Next, job enabled elasticity aka factor > 1 and checkpoints are written with taskName like "Parittion 0_0_2" and not taskname 0
3. Then after a while, job disables elasticity aka factor =1 again then we want to use checkpoints 0_0_2 to compute checkpoint of task with name "Partition 0". 

This config when true lets us know that we need to looking for checkpoints of other tasks even if the taksName does not match exactly. as in look for 0_0_2 and 0_1_2 to compute checkpoint of task 0.

without this config, we would have to infer that there was elasticity enabled in the past by looking at all checkpoints present in the stream. And this flow will apply to jobs even when elasticity was never enabled for the job. I wanted to avoid changes to the existing flow and guard all new changes behind a config and hence this.

Reply:
removed the config and using the existence of checkpoints with elastic names to trigger using all checkpoints for computing last proc offsets